src/java/org/apache/cassandra/index/sai/virtual/AnalyzerView.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.virtual;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+import com.google.common.base.Charsets;
+
+import org.apache.cassandra.db.DecoratedKey;
+import org.apache.cassandra.db.marshal.CompositeType;
+import org.apache.cassandra.db.marshal.UTF8Type;
+import org.apache.cassandra.db.virtual.AbstractVirtualTable;
+import org.apache.cassandra.db.virtual.SimpleDataSet;
+import org.apache.cassandra.dht.LocalPartitioner;
+import org.apache.cassandra.index.sai.analyzer.JSONAnalyzerParser;
+import org.apache.cassandra.index.sai.analyzer.LuceneAnalyzer;
+import org.apache.cassandra.schema.TableMetadata;
+import org.apache.cassandra.utils.ByteBufferUtil;
+import org.apache.lucene.analysis.Analyzer;
+
+public class AnalyzerView extends AbstractVirtualTable
```

Comment:
Can you provide an explanation how this works and what is in this virtual table?
I can understand from the code that there are 3 columns: `text`, `json_analyzer` and `tokens`. 
But who/what provides the data? Who/what provides the `partitionKey` for the `data` call?
Does it work the way I can just add a row with the first two columns and the `tokens` value gets automatically computed?

Reply:
Have you seen the unit test?

Reply:
Ok, the unit test explains that, thanks.